### PR TITLE
feature: gateway

### PR DESCRIPTION
**Pull request by geclos**
Implements Latitude API Gateway using Hono with the nodejs adapter

### CODE REVIEW

File: apps/gateway/test/setup.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// vitest-env.d.ts
+
+import useTestDatabase from '$core/tests/useTestDatabase'
+
+useTestDatabase()
```
Comment (andresgutgon) [inline apps/gateway/test/setup.ts:5]: Makes sense to put this here if we have to put inside describes anyway?

File: apps/gateway/test/vitest-env.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import 'vitest'
+
+import * as factories from './factories'
```
Comment (andresgutgon) [inline apps/gateway/test/vitest-env.d.ts:3]: I guess copy & pasta

File: apps/gateway/vitest.config.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import tsconfigPaths from 'vite-tsconfig-paths'
```
Comment (andresgutgon) [inline apps/gateway/vitest.config.ts:1]: No need, remember vitest already handle this nicely

File: apps/gateway/test/setup.ts (hunk 1 above)
Comment (geclos) [reply apps/gateway/test/setup.ts:5]: you don't need to put it in describe :)

File: packages/core/src/schema/models/apiKeys.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ export const apiKeys = latitudeSchema.table(
     token: uuid('token')
       .notNull()
       .unique()
-      .$defaultFn(() => crypto.randomUUID()),
+      .default(sql`gen_random_uuid()`),
```
Comment (csansoon) [inline packages/core/src/schema/models/apiKeys.ts:22]: Why this change?

File: apps/gateway/test/setup.ts (hunk 1 above)
Comment (andresgutgon) [reply apps/gateway/test/setup.ts:3]: Ah no. Ok nice

File: packages/core/src/schema/models/apiKeys.ts (hunk 4 above)
Comment (andresgutgon) [reply packages/core/src/schema/models/apiKeys.ts:22]: This is generated natively by postgreSQL

Comment (geclos) [reply packages/core/src/schema/models/apiKeys.ts:22]: better to let the db handle defaults when possible instead of our code.

File: apps/gateway/test/setup.ts (hunk 1 above)
Comment (geclos) [reply apps/gateway/test/setup.ts:3]: fixed web

**Review by andresgutgon — APPROVED** [review]
